Can't delombok java module

I'm trying to use the delombok plugin to generate javadocs and its not working. I think its not working because the delombok task itself isn't working.

My project is here: https://github.com/docling-project/docling-java/tree/main/docling-serve/docling-serve-api

It has a `module-info.java` and several `compileOnly` dependencies. These dependencies are exposed in the module as `requires static`. They need to be `compileOnly` dependencies (for various reasons).

When running `./gradlew clean :docling-serve-api:delombok --console=plain` I'm getting all these messages:

```console
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/module-info.java:1: error: file should be on source path, or on patch path for module
open module ai.docling.serve.api {
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/ExportDocumentResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/Document.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/Chunk.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/ChunkDocumentResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/options/HierarchicalChunkerOptions.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/options/ChunkerOptions.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/options/HybridChunkerOptions.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/HybridChunkDocumentRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/HierarchicalChunkDocumentRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.chunk.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/util/ValidationUtils.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.util;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/util/Utils.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.util;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/util/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson3DurationSerializer.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.serialization;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson2DurationSerializer.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.serialization;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/DurationSerializationFormat.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.serialization;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/package-info.java:4: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/health/HealthCheckResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.health;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/health/package-info.java:4: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/DoclingServeHealthApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/response/ClearResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.clear.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/response/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/request/ClearResultsRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.clear.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/request/ClearConvertersRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.clear.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/request/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/DoclingServeConvertApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/response/TaskStatus.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.task.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/response/TaskStatusMetadata.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.task.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/response/TaskStatusPollResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.task.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/response/package-info.java:4: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/request/TaskResultRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.task.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/request/TaskStatusPollRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.task.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/request/package-info.java:4: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/task/package-info.java:4: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/DoclingServeChunkApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/DoclingServeTaskApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/DoclingServeApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/DoclingServeClearApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/package-info.java:4: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/response/ErrorItem.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/response/ConvertDocumentResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/response/DocumentResponse.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.response;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/response/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/ConvertDocumentRequest.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/VlmModelApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/OcrEngine.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/ResponseFormat.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/ConvertDocumentOptions.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/VlmModelLocal.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/PictureDescriptionApi.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/TableFormerMode.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/PdfBackend.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/PictureDescriptionLocal.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/VlmModelType.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/OutputFormat.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/ImageRefMode.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/InputFormat.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/ProcessingPipeline.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.options;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/target/ZipTarget.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.target;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/target/InBodyTarget.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.target;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/target/PutTarget.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.target;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/target/Target.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.target;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/target/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/source/HttpSource.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.source;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/source/Source.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.source;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/source/FileSource.java:1: error: file should be on source path, or on patch path for module
package ai.docling.serve.api.convert.request.source;
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/source/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/package-info.java:1: error: file should be on source path, or on patch path for module
@NullMarked
^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/module-info.java:3: error: module not found: org.jspecify
  requires static org.jspecify;
                     ^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/module-info.java:8: error: module not found: com.fasterxml.jackson.core
  requires static com.fasterxml.jackson.core;
                                       ^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/module-info.java:9: error: module not found: com.fasterxml.jackson.databind
  requires static com.fasterxml.jackson.databind;
                                       ^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/module-info.java:10: error: module not found: tools.jackson.core
  requires static tools.jackson.core;
                               ^
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/ExportDocumentResponse.java:3: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in the unnamed module, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/Document.java:5: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/response/Chunk.java:6: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/options/HybridChunkerOptions.java:3: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/HybridChunkDocumentRequest.java:5: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/ConvertDocumentOptions.java:7: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/ConvertDocumentOptions.java:13: error: package com.fasterxml.jackson.databind.annotation is not visible
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
                                     ^
  (package com.fasterxml.jackson.databind.annotation is declared in module jackson.databind, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/PictureDescriptionLocal.java:5: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/PictureDescriptionApi.java:8: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/options/PictureDescriptionApi.java:14: error: package com.fasterxml.jackson.databind.annotation is not visible
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
                                     ^
  (package com.fasterxml.jackson.databind.annotation is declared in module jackson.databind, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/chunk/request/HierarchicalChunkDocumentRequest.java:5: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/util/Utils.java:13: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.NonNull;
                   ^
  (package org.jspecify.annotations is declared in the unnamed module, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/util/Utils.java:14: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson3DurationSerializer.java:7: error: package tools.jackson.core is not visible
import tools.jackson.core.JacksonException;
                    ^
  (package tools.jackson.core is declared in the unnamed module, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson3DurationSerializer.java:8: error: package tools.jackson.core is not visible
import tools.jackson.core.JsonGenerator;
                    ^
  (package tools.jackson.core is declared in the unnamed module, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson2DurationSerializer.java:8: error: package com.fasterxml.jackson.core is not visible
import com.fasterxml.jackson.core.JsonGenerator;
                            ^
  (package com.fasterxml.jackson.core is declared in the unnamed module, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson2DurationSerializer.java:9: error: package com.fasterxml.jackson.databind is not visible
import com.fasterxml.jackson.databind.BeanProperty;
                            ^
  (package com.fasterxml.jackson.databind is declared in module jackson.databind, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson2DurationSerializer.java:10: error: package com.fasterxml.jackson.databind is not visible
import com.fasterxml.jackson.databind.JsonSerializer;
                            ^
  (package com.fasterxml.jackson.databind is declared in module jackson.databind, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson2DurationSerializer.java:11: error: package com.fasterxml.jackson.databind is not visible
import com.fasterxml.jackson.databind.SerializerProvider;
                            ^
  (package com.fasterxml.jackson.databind is declared in module jackson.databind, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/serialization/Jackson2DurationSerializer.java:12: error: package com.fasterxml.jackson.databind.ser is not visible
import com.fasterxml.jackson.databind.ser.ContextualSerializer;
                                     ^
  (package com.fasterxml.jackson.databind.ser is declared in module jackson.databind, but module ai.docling.serve.api does not read it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/health/HealthCheckResponse.java:3: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/clear/response/ClearResponse.java:3: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/request/ConvertDocumentRequest.java:5: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
                   ^
  (package org.jspecify.annotations is declared in module ai.docling.core, which does not export it)
/Users/edeandre/workspaces/IntelliJ/docling-java/docling-serve/docling-serve-api/src/main/java/ai/docling/serve/api/convert/response/DocumentResponse.java:3: error: package org.jspecify.annotations is not visible
import org.jspecify.annotations.Nullable;
```

--- a/lombok-plugin/src/main/java/io/freefair/gradle/plugins/lombok/LombokPlugin.java
+++ b/lombok-plugin/src/main/java/io/freefair/gradle/plugins/lombok/LombokPlugin.java
@@ -116,10 +116,18 @@ public class LombokPlugin implements Plugin<Project> {
 
             delombokTaskProvider.configure(delombok -> {
                 delombok.getEncoding().set(compileTaskProvider.map(c -> c.getOptions().getEncoding()));
-                delombok.getClasspath().from(sourceSet.getCompileClasspath());
                 delombok.getInput().from(sourceSet.getJava().getSourceDirectories());
                 delombok.getSourcepath().from(compileTaskProvider.flatMap(c -> c.getOptions().getGeneratedSourceOutputDirectory()));
                 delombok.dependsOn(sourceSet.getJava().getBuildDependencies());
+
+                boolean hasModuleInfo = !sourceSet.getJava().getSourceDirectories()
+                        .getAsFileTree().matching(pattern -> pattern.include("**/module-info.java")).isEmpty();
+
+                if (hasModuleInfo) {
+                    delombok.getModulePath().from(sourceSet.getCompileClasspath());
+                } else {
+                    delombok.getClasspath().from(sourceSet.getCompileClasspath());
+                }
             });
         });
 

--- a/lombok-plugin/src/main/java/io/freefair/gradle/plugins/lombok/LombokPlugin.java
+++ b/lombok-plugin/src/main/java/io/freefair/gradle/plugins/lombok/LombokPlugin.java
@@ -121,7 +121,7 @@ public class LombokPlugin implements Plugin<Project> {
                 delombok.dependsOn(sourceSet.getJava().getBuildDependencies());
 
                 boolean hasModuleInfo = !sourceSet.getJava().getSourceDirectories()
-                        .getAsFileTree().matching(pattern -> pattern.include("**/module-info.java")).isEmpty();
+                        .getAsFileTree().matching(pattern -> pattern.include("module-info.java")).isEmpty();
 
                 if (hasModuleInfo) {
                     delombok.getModulePath().from(sourceSet.getCompileClasspath());

--- a/lombok-plugin/src/test/java/io/freefair/gradle/plugins/lombok/DelombokTest.java
+++ b/lombok-plugin/src/test/java/io/freefair/gradle/plugins/lombok/DelombokTest.java
@@ -14,6 +14,33 @@ import static org.assertj.core.api.Assertions.assertThat;
 public class DelombokTest extends AbstractPluginTest {
 
     @Test
+    public void jpmsDelombokTest() {
+        createGradleConfiguration()
+                .applyPlugin("java")
+                .applyPlugin("io.freefair.lombok")
+                .addCustomConfigurationBlock("delombok.target = file('src/main-delombok/java/')")
+                .write();
+
+        TypeSpec.Builder builder = TypeSpec.classBuilder("ModularUser")
+                .addModifiers(Modifier.PUBLIC);
+        builder.addField(FieldSpec.builder(String.class, "name", Modifier.PRIVATE).build());
+        builder.addField(FieldSpec.builder(String.class, "email", Modifier.PRIVATE).build());
+        builder.addAnnotation(AnnotationSpec.builder(lombok.Data.class).build());
+
+        createJavaClass("main", "io.freefair.gradle.plugins.lombok.test", builder.build());
+
+        createFile("src/main/java", "module-info.java")
+                .append("module io.freefair.gradle.plugins.lombok.test {\n")
+                .append("    requires static lombok;\n")
+                .append("}\n")
+                .write();
+
+        executeTask("delombok", "--debug");
+        String modularUser = readJavaClass("main-delombok", "io.freefair.gradle.plugins.lombok.test", "ModularUser");
+        assertThat(modularUser).doesNotContain("lombok.Data");
+    }
+
+    @Test
     public void simpleDelombokTest() {
         createGradleConfiguration()
                 .applyPlugin("java")

